app/presenters/licences/view-licence-bills.presenter.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+/**
+ * Formats data for the `/licences/{id}/bills` view licence bill page
+ * @module ViewLicenceBillsPresenter
+ */
+
+const { formatLongDate, formatMoney } = require('../base.presenter')
+
+/**
+ * Formats data for the `/licences/{id}/bills` view licence bill page
+ *
+ * @returns {Object} The data formatted for the view template
+ */
+function go (bills) {
+  return {
+    activeTab: 'bills',
+    bills: _formatBillsToTableRow(bills)
+  }
+}
+
+function _formatBillsToTableRow (bills) {
+  return bills.map((bill) => {
+    return {
+      billNumber: bill.invoiceNumber,
+      dateCreated: formatLongDate(new Date(bill.createdAt)),
+      account: bill.accountNumber,
+      runType: bill.billRun?.batchType,
```

Comment:
you're saying here that billRun may be undefined but not doing anything if it is from what I can see? will this result in undefined being output on the screen?

Reply:
When testing if `billRun` did not exists there would be an error thrown (crashing the server potentially) this is to cover that exception. 

But the code below should always return a `billRun` object. So i can remove the optional chaining if desired. 
```
    .withGraphFetched('billRun')
    .modifyGraph('billRun', (builder) => {
      builder.select(['batchType'])
    })
```

The  current result would be a empty as below. 

![Screenshot 2024-05-07 at 14 59 16](https://github.com/DEFRA/water-abstraction-system/assets/58443816/26529b97-7e30-43da-9386-5b153eb3ed1c)

I thought this would be better to have then not have.

Reply:
In think in this case i think it's nunjucks - https://mozilla.github.io/nunjucks/templating.html#variables

`If a value is undefined or null, nothing is displayed. The same behavior occurs when referencing undefined or null objects. The following all output nothing if foo is undefined: {{ foo }}, {{ foo.bar }}, {{ foo.bar.baz }}.`

But i am not 100%.

Reply:
You can always assume that a bill run exists. There would never be a scenario where a bill existed but the bill run it linked to didn't.

Reply:
Optional chaining removed